sdk/typespec/typespec_client_core/src/http/sanitizer.rs

```diff
@@ -0,0 +1,144 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+// cspell:ignore traceparent tracestate
+
+use crate::http::{headers::Headers, Url};
+use std::{collections::HashSet, sync::LazyLock};
+
+pub static DEFAULT_ALLOWED_HEADER_NAMES: LazyLock<HashSet<&'static str>> = LazyLock::new(|| {
+    [
+        "accept",
+        "cache-control",
+        "connection",
+        "content-length",
+        "content-type",
+        "date",
+        "etag",
+        "expires",
+        "if-match",
+        "if-modified-since",
+        "if-none-match",
+        "if-unmodified-since",
+        "last-modified",
+        "ms-cv",
+        "pragma",
+        "request-id",
+        "retry-after",
+        "server",
+        "traceparent",
+        "tracestate",
+        "transfer-encoding",
+        "user-agent",
+        "www-authenticate",
+        "x-ms-request-id",
+        "x-ms-client-request-id",
+        "x-ms-return-client-request-id",
+    ]
+    .iter()
+    .copied()
+    .collect()
+});
+
+pub static DEFAULT_ALLOWED_QUERY_PARAMETERS: LazyLock<HashSet<&'static str>> =
+    LazyLock::new(|| ["api-version"].iter().copied().collect());
+
+pub const REDACTED_PATTERN: &str = "REDACTED";
```

Comment:
Is this what most languages use? I know it's "Sanitized" for tests pretty universally, but just from a familiarity perspective, we should try to use whatever is most common. Not a big deal, though. I suspected my old "******" weren't common in our codebases anyway.

Reply:
Go and .Net use "REDACTED" if it matters.